### ISSUE TITLE
regeneratorRuntime / JS Strict Mode issue after 4.11 -> 4.12 update

### ISSUE THREAD

**Issue by xenobytezero** [strapi/strapi#17692]
### Required System information

- Node.js version: 16.20.1
- NPM version: 8.19.4
- Strapi version: 4.12.5
- Database: Postgres
- Operating system:
- Is your project Javascript or Typescript: Typescript

### Describe the bug

This bug is very difficult to give a reproduction for, this is more a question in case something immediately jumps out.

I have a content-manager plugin that uses a dynamic `import()` to load a chunk of JS from the server. After updating from Strapi 4.11.1 to 4.12.5 the exact same chunk of code being loaded in exactly same way is producing a CSP error exactly as described in the following posts.

https://stephencharlesweiss.com/unsafe-eval-regenerator-runtime
https://github.com/jaredpalmer/tsdx/issues/954
https://github.com/facebook/regenerator/issues/455

TL;DR my code somehow uses the [regenerator-runtime](https://github.com/facebook/regenerator/tree/main#readme) library, which has a section that can trip if the JS is run inside strict mode (see the post for more details). When this trips the code tries to do an `unsafe-eval` and triggers the CSP block, causing the error. The code does not trip in 4.11, but seems to in 4.12. I assume something has changed in the build process for the admin that switches strict mode on?

The only difference I can see is the code is the 4.11 -> 4.12 update. The code being executed is binary identical.

Looking through changelogs between the two versions I can't see anything immediately that would explain this, so this is more of a question if this can be easily explained by someone via the update.

If more information is needed I can provide.

**Comment by derrickmehaffy** [comment]
@xenobytezero TBH I'm not really sure what to classify this as, can you please provide some reproduction steps?

**Comment by xenobytezero** [comment]
I'm actually probably just going to close this. Been investigating all day and it looks like my version of this situation appears to be a mixture of plugins built in specific ways causing an issue that I didn't know was there until Strapi made a change. Sorry to clog up the feeds.

**Comment by derrickmehaffy** [comment]
> I'm actually probably just going to close this. Been investigating all day and it looks like my version of this situation appears to be a mixture of plugins built in specific ways causing an issue that I didn't know was there until Strapi made a change. Sorry to clog up the feeds.

All good, better safe than sorry!
